### ISSUE TITLE
508 Accessibility Check

### ISSUE THREAD

**Issue by rjmajma** [18F/openFEC#551]
Do automated 508 tests to see if there are any issues that we need to fix. Please catalogue them here. 

Later, we'll bring to Nick Bristow.

**Comment by rjmajma** [comment]
There's a dependency here based on the new comps. May have to happen towards the end of the sprint.

**Comment by msecret** [comment]
I ran the site through [pa11y](https://github.com/nature/pa11y). Here are the results separated by pages

/
red
 > WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.InputText.Name (2 occurences)
   This text input element does not have a name available to an accessibility API. Valid names are: label element, title attribute.
```
<input type="text" name="search" class="search-bar tt-input" placeholder="Enter a candidate name" autocomplete="off" spellcheck="false" dir="auto" style="position: relative; vertical-align: top;">
```
```
<input id="glossary-search" type="text" placeholder="Search for terms" class="tt-input" autocomplete="off" spellcheck="false" dir="auto" style="position: relative; vertical-align: top;">
```
 > WCAG2AA.Principle1.Guideline1_3.1_3_1.F68 (2 occurences)
   This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrapped around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
```
<input type="text" name="search" class="search-bar tt-input" placeholder="Enter a candidate name" autocomplete="off" spellcheck="false" dir="auto" style="position: relative; vertical-align: top;">
```
```
<input id="glossary-search" type="text" placeholder="Search for terms" class="tt-input" autocomplete="off" spellcheck="false" dir="auto" style="position: relative; vertical-align: top;">
```
 > WCAG2AA.Principle1.Guideline1_3.1_3_1.H42.2
   Heading tag found with no content. Text that is not intended as a heading should not be marked up with heading tags.
```
<h4><dt><span id="glossary-term" cl...</h4>
```
 > WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.Button.Name (2 occurences)
   This button element does not have a name available to an accessibility API. Valid names are: title attribute, element content.
```
<button id="glossary-toggle" data-slide="right" class="header-nav__link side-toggle"><i class="ti ti-help-alt"></i></button>
```
```
<button class="button--close u-float-right" id="hide-glossary"><i class="ti-close text--white"...</button>
```
 > WCAG2AA.Principle3.Guideline3_2.3_2_2.H32.2 (2 occurences)
   Form does not contain a submit button (input type="submit", input type="image", or button type="submit").
```
<form id="search" action="/"> <span class="twitt...</form>
```
```
<form id="large-search" action="/"> <label for="search" ...</form>
```
 > WCAG2AA.Principle3.Guideline3_1.3_1_1.H57.2
   The html element should have a lang or xml:lang attribute which describes the language of the document.

Yellow
 > WCAG2AA.Principle1.Guideline1_3.1_3_1.H49.I (3 occurences)
   Semantic markup should be used to mark emphasised or special text so that it can be programmatically determined.
```
<i class="ti ti-help-alt"></i>
```
 > WCAG2AA.Principle1.Guideline1_4.1_4_3.G145.BgImage (4 occurences)
   This element's text is placed on a background image. Ensure the contrast ratio between the text and all covered parts of the image are at least 3:1.
 > WCAG2AA.Principle1.Guideline1_3.1_3_1.H44.NotFormControl
   This label's "for" attribute contains an ID for an element that is not a form control. Ensure that you have entered the correct ID for the intended element.
```
<label for="search" class="text--hero">To get started, search for a fe...</label>
```

**Comment by rjmajma** [comment]
Hey @msecret, can you create a new issue that details what work has to be done based on the test you ran? Then close this one please.

**Comment by msecret** [comment]
Created two issues:
https://github.com/18F/openFEC/issues/585
https://github.com/18F/openFEC/issues/584

The #584 is for severe accessibility problems that should be fixed, and the #585 is for less sever problems that should eventually be fixed.
